$OpenBSD: patch-src_gui_kernel_qkeymapper_x11_cpp,v 1.1 2013/11/15 10:52:24 dcoppa Exp $

From 0c03af0d4d928bdbb32b09eedb1dba3ce59e5278 Mon Sep 17 00:00:00 2001
From: Gatis Paeglis <gatis.paeglis@digia.com>
Date: Sat, 31 Aug 2013 21:22:47 +0200
Subject: [PATCH] Revert "QTBUG-15319: fix shortcuts with secondary Xkb layout."

The change which attempted to fix QTBUG-15319 broke keyboard
shortcuts for non latin keyboard layouts.

This patch reverts QTBUG-15319 (f45cdeda8) since it caused a
regression.

Task-number: QTBUG-32908

Change-Id: I47d7984fa7986d5218d1f3ff1fc36d2ec67c9ba7
Reviewed-by: David Faure <david.faure@kdab.com>

--- src/gui/kernel/qkeymapper_x11.cpp.orig	Fri Jun  7 07:16:59 2013
+++ src/gui/kernel/qkeymapper_x11.cpp	Wed Nov 13 18:09:16 2013
@@ -282,12 +282,9 @@ QList<int> QKeyMapperPrivate::possibleKeysXKB(QKeyEven
 
     // first, translate key only using lock modifiers (there are no Qt equivalents for these, so we must
     // always use them when determining the baseKeySym)
-    // Note: the Xkb group to be used for the conversion keycode->keysym has to be given to
-    //       XkbLookupKeySym(). This information is contained in the bits 8 to 15 of xmodifiers.
-    //       See https://bugreports.qt-project.org/browse/QTBUG-15319 .
     KeySym baseKeySym;
     uint consumedModifiers;
-    if (!XkbLookupKeySym(X11->display, xkeycode, (xmodifiers & (0xff00 | LockMask | qt_num_lock_mask)),
+    if (!XkbLookupKeySym(X11->display, xkeycode, (xmodifiers & (LockMask | qt_num_lock_mask)),
                          &consumedModifiers, &baseKeySym))
         return QList<int>();
 
